code/framework/extension/src/main/java/io/cattle/platform/extension/impl/ExtensionManagerImpl.java

```diff
@@ -61,6 +51,21 @@
     }
 
     @Override
+    public Map<String, Object> map(String key) {
+        return getExtensionMapInternal(key);
+    }
+
+    protected ExtensionMap<String, Object> getExtensionMapInternal(String key) {
+        ExtensionMap<String, Object> map = extensionMaps.get(key);
+        if (map == null) {
+            Map<String, Object> inner = started ? getExtensionMapInternal(key) : Collections.<String, Object>emptyMap();
```

Comment:
See https://github.com/hibooboo2/cattle/blob/add_map/code/framework/extension/src/main/java/io/cattle/platform/extension/impl/ExtensionManagerImpl.java#L83 to see why i did this. But that is the right call. This is a recursive function.